assets/src/edit-story/app/media/utils/getResourceFromLocalFile.js

```diff
@@ -120,21 +129,48 @@ const getVideoResource = async (file) => {
   });
 };
 
+const createPlaceholderResource = (properties) => {
+  return createLocalResource({ ...properties, isPlaceholder: true });
+};
+
+const getPlaceholderResource = (file) => {
+  const type = getTypeFromMime(file.type);
+  const mimeType = type === 'image' ? 'image/png' : 'video/mp4';
+
+  // The media library requires resources with valid mimeType and dimensions.
+  return createPlaceholderResource({
+    type: type || 'image',
+    mimeType: mimeType,
+    src: '',
+    width: PAGE_WIDTH,
+    height: PAGE_HEIGHT,
+  });
```

Comment:
Should there be a poster attribuate if type is video? Maybe the source of the poster be an blob of transparent 1x1 pixel?

Reply:
Placeholder resources don't need sources and shouldn't require a poster either. They're just gray rectangles essentially.